src/instances/HeadcountInstance.ts

```diff
@@ -815,27 +852,110 @@ export class HeadcountInstance {
      * @param {string} [reason] The reason.
      * @private
      */
-    private stopAllIntervalsAndCollectors(reason?: string): void {
-        if (this._intervalsAreRunning) {
-            if (this._headcountInterval) {
-                clearInterval(this._headcountInterval);
-                this._headcountInterval = null;
-            }
+    private async stopAllIntervalsAndCollectors(reason?: string) {
+        this._logger.info(`${this._instanceInfo} Stopping all intervals and collectors for reason: ${reason ?? null}`);
+        this._intervalsAreRunning = false;
 
-            if (this._controlPanelInterval) {
-                clearInterval(this._controlPanelInterval);
-                this._controlPanelInterval = null;
-            }
+        this._controlPanelReactionCollector?.stop();
+        this._controlPanelReactionCollector = null;
+
+        this._headcountButtonCollector?.stop();
+        this._headcountButtonCollector = null;
+        return;
+    }
 
-            this._intervalsAreRunning = false;
+    
+    /**
+     * Starts the intervals, which periodically updates the headcount message and the control panel message.
+     * @return {boolean} Whether the intervals started.
+     * @private
+     */
+     private startIntervals(): boolean {
+        if (!this._headcountMsg || !this._controlPanelMsg) return false;
+        if (this._intervalsAreRunning || this._headcountStatus !== HeadcountStatus.HEADCOUNT_IN_PROGRESS) return false;
+        this._intervalsAreRunning = true;
+        this._logger.info(`${this._instanceInfo} Starting all intervals`);
+
+        this.updateControlPanel();
+        this.updateHeadcountPanel();
+        
+        return true;
+    }
+
+    /**
+     * Interval for control panel
+     */
+    private async updateControlPanel(){
+        this._logger.debug(`${this._instanceInfo} Control Panel Interval`);
+        /**
```

Comment:
Use `/* */` for multi-line comments, `//` for single-line comments, and `/** */` for documentation only.